notebooks/optimal-stopping-point/README.md

```diff
@@ -0,0 +1,30 @@
+# Optimal Stopping Point Prediction
+
+The aim of this ML problem is to predict an optimal stopping point for CI tests based on their test duration (runtimes). We perform an initial data analysis as well as feature engineering on the testgrid data. Furthermore, we also calculated the optimal stopping point by identifying the distribution of the test_duration values for different CI tests and comparing the distributions of passing/failing tests.
+
+## Dataset
+
+ In order to achieve the optimal stopping point, we would be looking into the  [testgrid](https://testgrid.k8s.io/) data for all the passing and failed tests and find the distribution type for the test_duration metric. The test_duration metric tracks the time it took for a test to complete its execution. We can visualize the distribution of the test_duration metric across various testgrid dashboards and jobs. Based on the distribution type identified, we can find a point after which the test has a higher probability of failing.
+
+## Feature Engineering
+
+After fetching the data, we approximate the distributions of the test_duration variable and also check its Goodness of fit for different TestGrid tests across all TestGrid dashboards and grids. Based on the type of distribution identified, we can calculate the probability of the test failing.
+
+We fetch data for all the passing and failed tests and find the distribution type for the `test_duration` metric. The `test_duration` metric tracks the time it took for a test to complete its execution. We can visualize the distribution of the test_duration metric across various testgrid dashboards and jobs. Based on the distribution type identified, we can find top two distributions based on betterment of fit and probability density plots are used to understand data distribution for a continuous variable and likelihood (or probability) of obtaining a range of values that the continuous variable can assume. The area under the curve contains the probabilities for the test duration values.
+
+   * [Probability To Fail notebook](../data-sources/TestGrid/metrics/probability_to_fail.ipynb)
+
+## Model Training
+
+After performing initial data analysis and caluculating the probability to fail, we train a machine learning model to find the optimal stopping for a given test based on their test duration(runtimes).
```

Comment:
Could we clarify on this section a bit? IIUC it's not exactly training a ML model in a "traditional" way, but rather trying out some distributions and seeing which fits best for pass/fail data, and then finding the point where P(fail)>P(pass), right?